cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/impl/CassandraSequenceHandler.java

```diff
@@ -0,0 +1,155 @@
+		primaryKeyName.add( "sequence_name" );
+		List<String> columnNames = new ArrayList<String>( 2 );
+		columnNames.add( "sequence_name" );
+		columnNames.add( "sequence_value" );
+		List<String> columnTypes = new ArrayList<String>( 2 );
+		columnTypes.add( "varchar" );
+		columnTypes.add( "bigint" );
+		datastoreProvider.createColumnFamilyIfNeeded( generatorsKey, primaryKeyName, columnNames, columnTypes );
+	}
+
+	private Long nextValueSelect(String tableName, String sequenceName) {
+
+		Statement select = select().column( "sequence_value" ).from( "\"" + tableName + "\"" )
+				.where( eq( "sequence_name", QueryBuilder.bindMarker() ) );
+
+		PreparedStatement preparedStatement = provider.getSession().prepare( select.toString() );
+		BoundStatement boundStatement = preparedStatement.bind( sequenceName );
+
+		ResultSet resultSet;
+		try {
+			resultSet = provider.getSession().execute( boundStatement );
+		}
+		catch (DriverException e) {
+			System.out.println( e.toString() );
+			throw e;
+		}
+
+		if ( resultSet.isExhausted() ) {
+			return null;
+		}
+		else {
+			return resultSet.one().getLong( 0 );
+		}
+	}
+
+	private Long nextValueInsert(String tableName, String sequenceName, Long value) {
+
+		Insert insert = insertInto( "\"" + tableName + "\"" )
+				.value( "sequence_name", QueryBuilder.bindMarker( "sequence_name" ) )
+				.value( "sequence_value", QueryBuilder.bindMarker( "sequence_value" ) )
+				.ifNotExists();
+
+		PreparedStatement preparedStatement = provider.getSession().prepare( insert.toString() );
+		BoundStatement boundStatement = preparedStatement.bind();
+		boundStatement.setString( "sequence_name", sequenceName );
+		boundStatement.setLong( "sequence_value", value );
+
+		try {
+			provider.getSession().execute( boundStatement );
+		}
+		catch (DriverException e) {
+			System.out.println( e.toString() );
+			throw e;
+		}
+
+		return nextValueSelect( tableName, sequenceName );
+	}
+
+	private boolean nextValueUpdate(String tableName, String sequenceName, Long oldValue, Long newValue) {
+
+		Statement update = update( "\"" + tableName + "\"" )
+				.with( set( "sequence_value", QueryBuilder.bindMarker( "sequence_value_new" ) ) )
+				.where( eq( "sequence_name", QueryBuilder.bindMarker( "sequence_name" ) ) )
+				.onlyIf( eq( "sequence_value", QueryBuilder.bindMarker( "sequence_value_old" ) ) );
+
+		PreparedStatement preparedStatement = provider.getSession().prepare( update.toString() );
+		BoundStatement boundStatement = preparedStatement.bind();
+		boundStatement.setString( "sequence_name", sequenceName );
+		boundStatement.setLong( "sequence_value_new", newValue );
+		boundStatement.setLong( "sequence_value_old", oldValue );
+
+		ResultSet resultSet;
+		try {
+			resultSet = provider.getSession().execute( boundStatement );
+		}
+		catch (DriverException e) {
+			System.out.println( e.toString() );
+			throw e;
+		}
+
+		return resultSet.one().getBool( 0 );
+	}
+
+	public Number nextValue(NextValueRequest request) {
```

Comment:
Counters in cassandra are somewhat weird beasts that don't work the same way as everything else. In particular, you can't do CAS ('lightweight transactions' in cassandra speak) on them. It's like an AtomicLong you can't put a synchronised block around, so you can't read and update in a way that would allow you to build seq ids on them.  UUID is really the way to go for creating synthetic id values in cassandra, I don't expect many users to utilise the equivalent of 'auto generate col seq num' approach as they would in a relational database, but at least with this clunky approach they have the option if they really want to.